applications/tari_dan_wallet_cli/src/command/transaction.rs

```diff
@@ -260,13 +260,19 @@ pub async fn handle_submit(args: SubmitArgs, client: &mut WalletDaemonClient) ->
         });
     }
 
-    let request = TransactionSubmitRequest {
-        signing_key_index: None,
-        fee_instructions: vec![Instruction::CallMethod {
+    let fee_instructions = if common.dry_run {
+        vec![]
```

Comment:
Updated the dry run to automatically simulate fees if there is any `fee_instructions` in the transaction, which should include the required amount as well. 

IMO for now it's good enough. It would be better to simulate it always regardless of `fee_instructions`, but it would require a lot of refactor inside the transaction processor and tracker.

Reply:
Thanks LGTM